### PR TITLE
SLM-12 verify link service

### PR DESCRIPTION
**Pull request by mikehalmamoj**
And some tidying

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/sendlegalmailtoprisonsapi/config/SendLegalMailToPrisonsApiExceptionHandler.kt [hunk 1]
```diff
@@ -95,5 +95,6 @@ enum class ErrorCode(val userMessage: String) {
   NOT_FOUND("Not found"),
   // Custom errors
   EMAIL_MANDATORY("The email address must be entered"),
-  INVALID_CJSM_EMAIL("The email is not a recognised CJSM email address"),
+  INVALID_EMAIL("Enter an email address in the correct format"),
+  INVALID_CJSM_EMAIL("Enter an email address which ends with 'cjsm.net'"),
```
Comment (mikehalmamoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/sendlegalmailtoprisonsapi/config/SendLegalMailToPrisonsApiExceptionHandler.kt:99]: Aligned with copy from designers

File: src/main/kotlin/uk/gov/justice/digital/hmpps/sendlegalmailtoprisonsapi/security/JwtService.kt [hunk 2]
```diff
@@ -41,25 +39,24 @@ class JwtService(jwtConfig: JwtConfig) {
     Jwts.builder()
       .setId(UUID.randomUUID().toString())
       .setSubject(email)
-      .setExpiration(
-        Date.from(
-          LocalDateTime.now().plus(expiry.toMillis(), ChronoUnit.MILLIS).toInstant(zoneOffset())
-        )
-      )
+      .setExpiration(Date.from(Instant.now().plus(expiry.toMillis(), ChronoUnit.MILLIS)))
+      .addClaims(mapOf("authorities" to arrayOf("ROLE_SLM_CREATE_BARCODE")))
```
Comment (mikehalmamoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/sendlegalmailtoprisonsapi/security/JwtService.kt:43]: Missed from previous PR - the JWT should have a role assigned

File: src/main/kotlin/uk/gov/justice/digital/hmpps/sendlegalmailtoprisonsapi/security/JwtService.kt [hunk 3]
```diff
@@ -41,25 +39,24 @@ class JwtService(jwtConfig: JwtConfig) {
...
+        log.warn("Found an invalid JWT: $jwt", it)
       }
     }.isSuccess
 
   fun subject(jwt: String): String =
     Jwts.parser().setSigningKey(publicKey).parseClaimsJws(jwt).body.subject
+
+  @Suppress("UNCHECKED_CAST")
+  fun authorities(jwt: String): List<String>? =
+    Jwts.parser().setSigningKey(publicKey).parseClaimsJws(jwt).body["authorities"] as? List<String>
```
Comment (mikehalmamoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/sendlegalmailtoprisonsapi/security/JwtService.kt:61]: The type of body is a Map<String, Object> but we know that if the authorities are there then they will be a list

File: src/test/kotlin/uk/gov/justice/digital/hmpps/sendlegalmailtoprisonsapi/integration/magiclink/MagicLinkResourceTest.kt [hunk 4]
```diff
@@ -73,7 +73,22 @@ class MagicLinkResourceTest(
         .expectStatus().isBadRequest
         .expectBody()
         .jsonPath("$.errorCode").isEqualTo(ErrorCode.EMAIL_MANDATORY.name)
-        .jsonPath("$.userMessage").isEqualTo("The email address must be entered")
+        .jsonPath("$.userMessage").value<String> { it.contains("email address") }
```
Comment (mikehalmamoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/sendlegalmailtoprisonsapi/integration/magiclink/MagicLinkResourceTest.kt:76]: Don't test copy - just look for key words

**Review by yasinmustafa — APPROVED** [review]
LGTM

**Pull request merged**